client/src/components/EditProfile/EditPhoto/EditPhoto.tsx

```diff
@@ -1,81 +1,90 @@
-import { useState } from 'react';
+import { ChangeEventHandler, useState } from 'react';
 import { Paper, Box, Grid, Button, IconButton, Avatar, Typography } from '@material-ui/core';
 import useStyles from './useStyles';
 import DeleteSharpIcon from '@material-ui/icons/DeleteSharp';
 import { uploadImage } from '../../../helpers/APICalls/uploadImage';
+import editProfile from '../../../helpers/APICalls/editProfile';
+import { useAuth } from '../../../context/useAuthContext';
+import { useSnackBar } from '../../../context/useSnackbarContext';
 
 export default function EditPhoto(): JSX.Element {
-  const [file, setFile] = useState<string>('');
+  const { loggedInUserProfile, fetchProfileAndUpdateContext } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
 
   const classes = useStyles();
 
-  const handleCapture = ({ target }: any) => {
+  const handleCapture = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
     const files = target.files;
     if (files) {
-      setFile(URL.createObjectURL(files[0]));
       uploadImage({ file: files[0] })
-        .then()
+        .then(() => {
+          fetchProfileAndUpdateContext();
+          updateSnackBarMessage('Your photo is uploaded successfully');
+        })
         .catch(() => ({
           error: { message: 'Unable to connect to server. Please try again' },
         }));
     }
   };
 
-  function removeFile() {
-    setFile('');
+  async function removeFile() {
+    const profile = loggedInUserProfile;
+    if (profile) {
+      profile.photo = '';
+      await editProfile(profile);
```

Comment:
Everytime you use the `await` keyword, you should be aware that there is a chance for it to `throw` ([ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description))! Maybe we should wrap this in a `try...catch` block 👍🏼

Reply:
Good reminder. Done.